misc/ballerina-format/src/main/java/org/ballerinalang/format/FormatUtil.java

```diff
@@ -0,0 +1,387 @@
+import org.ballerinalang.launcher.BLauncherCmd;
+import org.ballerinalang.launcher.LauncherUtils;
+import org.wso2.ballerinalang.compiler.Compiler;
+import org.wso2.ballerinalang.compiler.tree.BLangCompilationUnit;
+import org.wso2.ballerinalang.compiler.tree.BLangPackage;
+import org.wso2.ballerinalang.compiler.tree.BLangTestablePackage;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+import org.wso2.ballerinalang.compiler.util.CompilerOptions;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Scanner;
+
+import static org.ballerinalang.compiler.CompilerOptionName.COMPILER_PHASE;
+import static org.ballerinalang.compiler.CompilerOptionName.EXPERIMENTAL_FEATURES_ENABLED;
+import static org.ballerinalang.compiler.CompilerOptionName.LOCK_ENABLED;
+import static org.ballerinalang.compiler.CompilerOptionName.OFFLINE;
+import static org.ballerinalang.compiler.CompilerOptionName.PRESERVE_WHITESPACE;
+import static org.ballerinalang.compiler.CompilerOptionName.PROJECT_DIR;
+import static org.ballerinalang.compiler.CompilerOptionName.SIDDHI_RUNTIME_ENABLED;
+import static org.ballerinalang.compiler.CompilerOptionName.SKIP_TESTS;
+import static org.ballerinalang.compiler.CompilerOptionName.TEST_ENABLED;
+import static org.ballerinalang.langserver.compiler.format.TextDocumentFormatUtil.generateJSON;
+
+/**
+ * Util class for compilation and format execution for formatting CLI tool.
+ */
+public class FormatUtil {
+    static final String CMD_NAME = "format";
+    private static final PrintStream outStream = System.err;
+
+    /**
+     * Execute formatter.
+     *
+     * @param argList        argument list from the console
+     * @param overWriteFile  file overwrite confirmation
+     * @param helpFlag       flag to get the help page
+     * @param sourceRootPath execution path
+     */
+    static void execute(List<String> argList, boolean overWriteFile, boolean helpFlag, Path sourceRootPath) {
+        if (helpFlag) {
+            String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(CMD_NAME);
+            outStream.println(commandUsageInfo);
+            return;
+        }
+
+        if (argList != null && argList.size() > 1) {
+            throw LauncherUtils.createUsageExceptionWithHelp(Messages.getArgumentError());
+        }
+
+        String moduleName;
+        String ballerinaFilePath;
+
+        try {
+            // If parameters are available user has given either the module name or the ballerina file path.
+            // Else user is in a ballerina project and expecting to format the whole ballerina project
+            if (argList != null && argList.size() > 0) {
```

Comment:
Seems like some nested blocks can be removed. Shall we check and remove them to reduce the number of nested blocks?

Reply:
Will check and remove...